client/blocks/app-banner/test/app-banner.jsx

```diff
@@ -0,0 +1,63 @@
+/**
+ * @format
+ * @jest-environment jsdom
+ */
+
+/**
+ * Internal dependencies
+ */
+import { getiOSDeepLink, buildDeepLinkFragment } from 'blocks/app-banner';
+import { EDITOR, NOTES, READER, STATS } from 'blocks/app-banner/utils';
+
+describe( 'iOS deep link fragments', () => {
+	test( 'properly encodes tricky fragments', () => {
+		expect( buildDeepLinkFragment( '?://&#', STATS ) ).toEqual( '%3F%3A%2F%2F%26%23' );
```

Comment:
even though I suggested writing this code, I wonder how much this ends up conflating the tests here. we might consider using this instead…

```js
const funnyFragment = '?://&#';

expect( buildDeepLinkFragment( funnyFragment, STATS ) ).toEqual( encodeURIComponent( funnyFragment ) );
```

that way the test isn't dependent on the implementation of `encodeURIComponent()` (not that it will change any time soon) and it also indicates in the test code the intended behavior more semantically - we get the expected intention instead of the expected outcome